src/lib/GSENSOR/devGsensor.cpp

```diff
@@ -35,24 +44,58 @@ static int start()
 
 static int timeout()
 {
-    gsensor.handle();
-
-    system_quiet_state = gsensor.getSystemState();
-    //When system is idle, set power to minimum
-    if(config.GetMotionMode() == 1)
+    static unsigned long lastCall = 0;
+    unsigned long now = millis();
+    if (gsensor.handleBump(now))
     {
-        if((system_quiet_state == GSENSOR_SYSTEM_STATE_QUIET) && (system_quiet_pre_state == GSENSOR_SYSTEM_STATE_MOVING) && !IsArmed())
+        if (bumps == 0)
         {
-        POWERMGNT::setPower(MinPower);
+            firstBumpTime = now;
         }
-        if((system_quiet_state == GSENSOR_SYSTEM_STATE_MOVING) && (system_quiet_pre_state == GSENSOR_SYSTEM_STATE_QUIET))
+        lastBumpTime = now;
+        bumps++;
+    }
+    if (bumps > 0 && (now - lastBumpTime > 400 || now - firstBumpTime > 1000))
+    {
+        float x, y, z;
+        gsensor.getGSensorData(&x, &y, &z);
+        if (bumps == 1 && fabs(x) < 0.5 && y < -0.8 && fabs(z) < 0.5)   // Single bump while holding the radio antenna up
         {
-        POWERMGNT::setPower((PowerLevels_e)config.GetPower());
+            if (connectionState == connected)
+            {
+                DBGLN("Loaning model");
+                SendRxLoanOverMSP();
+            }
+            else
+            {
+                DBGLN("Borrowing model");
+                // defer this calling `EnterBindingMode` for 2 seconds
+                deferExecution(2000, EnterBindingMode);
+            }
```

Comment:
Should this somehow be protected to prevent borrowing my own loan? 2 bumps 400ms apart mean I both loan and borrow my own model. 
```

UART STATS Bad:Good = 0:500
UART STATS Bad:Good = 0:500
UART STATS Bad:Good = 0:500
Loaning model
Bumps 1 : 0.183 -1.87 0.193  // <-- my bump

UART STATS Bad:Good = 0:500
Borrowing model
Bumps 1 : 0.33 0.974 0.3 // <-- putting down the controller

UART STATS Bad:Good = 0:500
UART STATS Bad:Good = 0:500
hwTimer stop
set rate 5
hwTimer interval: 20000
Config LoRa
Adjusted max packet size 64-64
hwTimer resume
Entered binding mode at freq = 12302619
$X<     /¤Îw!set rate 2
hwTimer interval: 2000
Config LoRa
Adjusted max packet size 32-32
Exiting binding mode
UART STATS Bad:Good = 0:428
UART STATS Bad:Good = 0:455
UART STATS Bad:Good = 0:500
```

Reply:
So what do you think about putting a stop to bump detection for 10s (or some other time period) after entering loan mode by bumping.